files/static/index.js

```diff
@@ -0,0 +1,340 @@
+        this.setTheme("theme-green");
+      } else {
+        $("#summary").textContent = "" + domain + " has " + (checkDomainIssues.warnings.length == 1 ? "a warning" : "warnings") + ", but satisfies all requirements for preloading.";
+        this.setTheme("theme-yellow");
+      }
+    } else {
+        $("#summary").textContent = "" + domain + " has errors and cannot be preloaded.";
+        this.setTheme("theme-red");
+    }
+
+    switch (domainStatus.status) {
+    case "unknown":
+      if (checkDomainIssues.errors.length === 0) {
+        $("#status").textContent = "" + domain + " is not yet preloaded, and may be submitted to the preload list."
+      } else {
+        $("#status").textContent = "" + domain + " is not preloaded."
+      }
+      break;
+    case "pending":
+      $("#status").textContent = "" + domain + " is pending submission to the preload list."
+      break;
+    case "preloaded":
+      $("#status").textContent = "" + domain + " is currently preloaded."
+      break;
+    case "rejected":
+      // TODO: do something useful if domainStatus.message is not present.
+      if (domainStatus.message) {
+        $("#status").textContent = "" + domain + " was rejected from the preload list for the following reason: " + domainStatus.message;
+      } else {
+        $("#status").textContent = "" + domain + " was rejected from the preload list.";
+      }
+      if (checkDomainIssues.errors.length === 0) {
+        $("#status").textContent += " It may be submitted again."
+      }
+      break;
+    case "removed":
+      if (checkDomainIssues.warnings.length === 0) {
+        $("#status").textContent = "" + domain + " was previously on the preload list, but has been removed. It may be submitted again.";
+      } else {
+        $("#status").textContent = "" + domain + " was previously on the preload list, but has been removed.";
+      }
+      break;
+    default:
+      $("#status").textContent = "Cannot determine preload status.";
+    }
+
+    function createIssueElement(issue, bullet, className) {
+      var el = document.createElement("div");
+      el.classList.add(className);
+      // summary.title = issue.code;
+
+      var bulletSpan = document.createElement("span");
+      bulletSpan.textContent = bullet + " ";
+      el.appendChild(bulletSpan);
+
+      var summary = document.createElement("span")
+      summary.classList.add("summary");
+      summary.textContent = issue.summary;
+      el.appendChild(summary);
+
+      var br = document.createElement("br")
+      el.appendChild(br);
+
+      var message = document.createElement("span")
+      message.textContent = issue.message;
+      el.appendChild(message);
+
+      return el;
+    }
+
+    $("#errors").textContent = "";
+    $("#warnings").textContent = "";
+    for (var i in checkDomainIssues.errors) {
+      var error = checkDomainIssues.errors[i];
+      var el = createIssueElement(error, "❌", "error");
```

Comment:
nit (more of a question): Hmm, I wonder if you should escape these characters and use \unnnn versions. I can't seem to find any references to it in internal code search.